### PR TITLE
Replace troublesome unicode quote characters with more friendly ones

### LINKED ISSUE #3128: LC_CTYPE=en_US.UTF-8 causes warnings in sct_download_data
Seen in https://github.com/neuropoly/spinalcordtoolbox/runs/1606746772?check_suite_focus=true and again in https://github.com/neuropoly/spinalcordtoolbox/runs/1608360364?check_suite_focus=true. Only on CentOS on Github Actions? Which is weird...because it's a docker image so it should be identical to Travis :haircut_man: ?

There's some locale fiddliness. Given that Travis sets `LC_ALL=en_US.UTF-8`, but not `LC_CTYPE=en_US.UTF-8` like Github Actions does, I suspect that's the problem.

```
Trying URL: https://github.com/sct-data/PAM50/releases/download/r20201104/PAM50-r20201104.zip
Downloading: PAM50-r20201104.zip
--- Logging error ---
Traceback (most recent call last):
  File "/github/home/sct_dev/python/envs/venv_sct/lib/python3.6/logging/__init__.py", line 996, in emit
    stream.write(msg)
UnicodeEncodeError: 'ascii' codec can't encode character '\u201c' in position 42: ordinal not in range(128)
Call stack:
  File "/github/home/sct_dev/spinalcordtoolbox/scripts/sct_download_data.py", line 168, in <module>
    res = main()
  File "/github/home/sct_dev/spinalcordtoolbox/scripts/sct_download_data.py", line 160, in main
    install_data(url, dest_folder, keep=arguments.k)
  File "/github/home/sct_dev/spinalcordtoolbox/download.py", line 141, in install_data
    logger.warning("Removing existing destination folder \u201c%s\u201d", dest_folder)
Message: 'Removing existing destination folder \u201c%s\u201d'
Arguments: ('/github/home/sct_dev/data/PAM50',)
```

I haven't tried to reproduce this yet. It should be relatively easy though:

```
LC_CTYPE=en_US.UTF-8 sct_download_data -d PAM50
```

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

`“` causes issues. Replace with `'`.

## Linked issues

Part of fixing #3128.

### CODE REVIEW

**Review by kousu — CHANGES_REQUESTED** [review]
Should we do these other cases too?

```
[kousu@requiem spinalcordtoolbox]$ git grep "“"
documentation/source/overview/concepts/quality-control.rst:These reports consist in “appendable” HTML files, containing a table
documentation/source/overview/concepts/spaces-and-coordinates.rst:images, have their contents indexed in “Fortran order”, meaning that in
documentation/source/overview/concepts/spaces-and-coordinates.rst:An “image orientation” corresponds to the orientation of the
documentation/source/overview/concepts/spaces-and-coordinates.rst:- `Matlab FieldTrip toolbox “How are the different head and MRI coordinate systems defined?”
spinalcordtoolbox/download.py:        logger.warning("Removing existing destination folder “%s”", dest_folder)
spinalcordtoolbox/download.py:                logger.warning("Updating existing “%s”", dstpath)
spinalcordtoolbox/registration/register.py:    Finds the angle of an image based on the method described by Sun, “Symmetry Detection Using Gradient Information.”
```

We should definitely also fix the second spinalcordtoolbox/download.py one; the others are in docs it looks like but, still maybe good to get them out?

**Comment by joshuacwnewton** [comment]
Good thinking. :sweat_smile:

**Comment by kousu** [comment]
Oh I just realized this maybe wasn't quite the crash I thought it was: https://github.com/neuropoly/spinalcordtoolbox/issues/3128#issuecomment-751822864

Still, getting these out will make debugging the actual crashes easier.

**Comment by joshuacwnewton** [comment]
> Oh I just realized this maybe wasn't quite the crash I thought it was: [#3128 (comment)](https://github.com/neuropoly/spinalcordtoolbox/issues/3128#issuecomment-751822864)
> 
> Still, getting these out will make debugging the actual crashes easier.

Thanks for letting me know. I've unlinked the issue for now. :slightly_smiling_face:

File: spinalcordtoolbox/registration/register.py [hunk 1]
```diff
@@ -1234,7 +1234,7 @@ def find_index_halfmax(data1d):
 
 def find_angle_hog(image, centermass, px, py, angle_range=10):
     """
-    Finds the angle of an image based on the method described by Sun, “Symmetry Detection Using Gradient Information.”
+    Finds the angle of an image based on the method described by Sun, 'Symmetry Detection Using Gradient Information.'
```
Comment (kousu) [inline spinalcordtoolbox/registration/register.py:1237]:
```suggestion
    Finds the angle of an image based on the method described by Sun, "Symmetry Detection Using Gradient Information."
```

**Review by kousu — CHANGES_REQUESTED** [review]
Just nitpicking.

I think `'` is good for the code since I think that's what we standardized on everywhere else, but the docs/comments are trying to follow academic conventions.

File: spinalcordtoolbox/registration/register.py (hunk 1 above)
Comment (joshuacwnewton) [reply spinalcordtoolbox/registration/register.py:1237]: Ah right. Using `'` in strings is so habitual that I didn't notice this was triple-quoted. :p

**Review by kousu — APPROVED** [review]
Thanks so much for doing the leg-work @joshuacwnewton

**Pull request merged**